src/coreclr/jit/llvm.cpp

```diff
@@ -493,6 +481,35 @@ CorInfoType Llvm::getCorInfoTypeForArg(CORINFO_SIG_INFO& sigInfo, CORINFO_ARG_LI
     return strip(corTypeWithMod);
 }
 
+FunctionType* Llvm::getFunctionType()
```

Comment:
Hmm, shouldn't we call it in `Llvm::Compile` (right now it is unused)?

Nit: I would name it something like `createFunctionTypeForCompilation`, to signify it is to be called only once (and is meant to be used for the function being compiled).

Reply:
Ah sorry, I now realise there is a problem with this method in that it can't distinguish between `TYP_INT` and `TYP_I_IMPL` (being a macro to `TYP_INT`, and my `i8*` args end up being `i32` ,  `toCorInfoType` maps them both to `CORINFO_TYPE_INT`.  Is it ok to use `lvIsPtr` to mark the actual pointers ?

Reply:
No... it is only used by the GS security code for some very specific purposes.

What you can do it is add some more metadata, that preserves "the original type". Look for `lvaInitVarDsc` in `lclvars.cpp`.

Reply:
I was writing this assuming we need the actual full type, not just `i8*` for pointers. If we don't, then `lvIsPtr` should be enough.

Reply:
Ultimately, we are going to need to build `IDynamicCastableGetInterfaceImplementation(i8* %0, %"[S.P.CoreLib]Internal.Runtime.MethodTable"* %1, i16 %2)`  so I think we'll need the full type.  I don't know that we are currently covering any of these signatures currently.